cmake/modules/FindOmrCuda.cmake

```diff
@@ -0,0 +1,66 @@
+###############################################################################
+# Copyright (c) 2019, 2020 IBM Corp. and others
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which accompanies this
+# distribution and is available at http://eclipse.org/legal/epl-2.0
+# or the Apache License, Version 2.0 which accompanies this distribution
+# and is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# This Source Code may also be made available under the following Secondary
+# Licenses when the conditions for such availability set forth in the
+# Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+# version 2 with the GNU Classpath Exception [1] and GNU General Public
+# License, version 2 with the OpenJDK Assembly Exception [2].
+#
+# [1] https://www.gnu.org/software/classpath/license.html
+# [2] http://openjdk.java.net/legal/assembly-exception.html
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+#############################################################################
+
+# This module locates a minimal set of cuda headers, required by the OMR_OPT_CUDA flag.
+# This module will search for CUDA resources in only one directory. The variables tested, in order of preference, are:
+#
+# 1. The cmake variable OMR_CUDA_HOME, if set.
+# 2. The cmake variable CUDA_HOME, if set.
+# 3. The environment variable CUDA_HOME, if set.
+#
+
+include(FindPackageHandleStandardArgs)
+
+set(OMR_CUDA_HOME "NOTFOUND" CACHE STRING "Path to the CUDA SDK. Takes precedence over CUDA_HOME in OMR.")
+set(OmrCuda_INCLUDE_DIRS "OmrCuda_INCLUDE_DIRS-NOTFOUND")
+
+# Establish which directory we are searching in
+
+set(OmrCuda_SEARCH_DIR "")
+
+if(OMR_CUDA_HOME)
+	set(OmrCuda_SEARCH_DIR "${OMR_CUDA_HOME}")
+elseif(CUDA_HOME)
+	set(OmrCuda_SEARCH_DIR "${CUDA_HOME}")
+elseif(ENV{CUDA_HOME})
+	set(OmrCuda_SEARCH_DIR "$ENV{CUDA_HOME}")
+else()
+	message(WARNING "CUDA support requested, but OMR_CUDA_HOME/CUDA_HOME are not set.")
```

Comment:
Should this not be an error?

Reply:
I was torn. If we make this an error, and the user doesn't specify `REQUIRED` in `find_package(OmrCuda REQUIRED)`, then we will have a hard error when there shouldn't be. However, this package doesn't function as a traditional "find package" module, and I find it hard to believe anyone would use this package without `OMR_OPT_CUDA`. Making this an error is probably fine.

If we leave this as a warning, the build will still error out when it's misconfigured, because OMR makes `OmrCuda` `REQUIRED` when `OMR_OPT_CUDA` is enabled.

I could go either way, so let me know what you prefer.

Reply:
I forgot about the  (optional)`REQUIRED` argument to `FindOmrCuda`. I think it should be an error in the required case and I'm fine with a warning otherwise.

Reply:
Done 👍